src/plugins/memdump/memdump.cpp

```diff
@@ -114,6 +114,40 @@
 
 #define DUMP_NAME_PLACEHOLDER "(not configured)"
 
+static void save_file_metadata(const drakvuf_trap_info_t* info,
+                               const char* file_path,
+                               size_t dump_size,
+                               addr_t dump_address,
+                               const char* method,
+                               const char* dump_reason,
+                               extras_t* extras)
+{
+    char* file = NULL;
+    if ( asprintf(&file, "%s.metadata", file_path) < 0 )
```

Comment:
Unfortunately, this drops the dump-deduplication feature. You will end up with a lot of files which do have exactly the same contents :( The previous naming scheme was resistant to this issue, because dumps with matching base address and contents hash were getting the exact same file name each time.

Also - right now it is not possible to determine the image base easily. For us this was the only data that was really necessary to perform dumps further. We apply Yara signatures against the dumps, so basically we can guess what's inside the dump just based on the pure contents. Dump base address is super helpful for disassembling, which is what people do both manually, semiautomatically and automatically (and in each case, they need to know the base address quickly).

Proposition 1:
So. How about a compromise solution of writing the metadata file in your format, while keeping the original dump file names? You could add something like `dump filename` to the metadata file, in order to link everything together. In this case, I can even write a Python script for you, which would traverse the `dumps/` directory, read the metadata and rename files to match your naming scheme, so you still would have the possibility to work with them easily using your setup.

Proposition 2:
Changing the dump file names to `<zero padded counter>_<base address>_<content hash>.dmp` and then e.g. in bash you can easily refer to them just by the counter value, like `0001_*.dmp`, while keeping everything +/- compatible with the existing situation.

I would prefer not to branch into too many different formats supported in parallel, because someday it may become harder to maintain.

Reply:
@icedevml thank you very much.

I will check some ideas and will back with some proposition (may be agree with proposition 1).

> In this case, I can even write a Python script for you

Thank you! Though I could do it myself too :)

Reply:
@icedevml, I have stopped on proposition 1. If You accept this we could merge the PR.